component/authenticator/src/test/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticatorTest.java

```diff
@@ -0,0 +1,299 @@
+
+        //Pattern configured no matching in subjectDN
+        X509Certificate cert4 = (X509Certificate) factory
+                .generateCertificate(new ByteArrayInputStream(DatatypeConverter.parseBase64Binary(CERTIFICATE_4)));
+        X509Certificate obj4[] = { cert4, null };
+
+        SequenceConfig sequenceConfig4 = new SequenceConfig();
+        Map<Integer, StepConfig> stepMap4 = new HashMap<>();
+        StepConfig stepConfig4 = new StepConfig();
+        stepConfig4.setAuthenticatedUser(null);
+        stepMap4.put(1, stepConfig4);
+        sequenceConfig4.setStepMap(stepMap4);
+
+        AuthenticatorConfig authenticatorConfig4 = new AuthenticatorConfig();
+        Map<String, String> parameterMap4 = new HashMap<String, String>();
+        parameterMap4.put(X509CertificateConstants.USERNAME, "CN");
+        parameterMap4.put(X509CertificateConstants.USER_NAME_REGEX, "^[a-zA-Z]{3}.[a-zA-Z]{2}$");
+        authenticatorConfig4.setParameterMap(parameterMap4);
+
+        //Authenticate using username attribute when pattern is configured
+        AuthenticatorConfig authenticatorConfig5 = new AuthenticatorConfig();
+        Map<String, String> parameterMap5 = new HashMap<String, String>();
+        parameterMap5.put(X509CertificateConstants.USERNAME, "CN");
+        parameterMap5.put(X509CertificateConstants.USER_NAME_REGEX, "\\d\\d\\d[a-zA-Z]{8}");
+        authenticatorConfig5.setParameterMap(parameterMap5);
+
+        //Authenticate using email attribute when pattern is configured
+        X509Certificate cert6 = (X509Certificate) factory
+                .generateCertificate(new ByteArrayInputStream(DatatypeConverter.parseBase64Binary(CERTIFICATE_5)));
+        X509Certificate obj6[] = { cert6, null };
+        AuthenticatorConfig authenticatorConfig6 = new AuthenticatorConfig();
+        Map<String, String> parameterMap6 = new HashMap<String, String>();
+        parameterMap6.put(X509CertificateConstants.USERNAME, "EMAILADDRESS");
+        parameterMap6.put(X509CertificateConstants.USER_NAME_REGEX, "[a-zA-Z]{3}\\d");
+        authenticatorConfig6.setParameterMap(parameterMap6);
+
+        return new Object[][] {
+                {
+                        obj1, authenticatorConfig1, sequenceConfig1, true,
+                        X509CertificateConstants.X509_CERTIFICATE_ALTERNATIVE_NAMES_NOTFOUND_ERROR
+                }, { obj2, authenticatorConfig2, sequenceConfig2, false, "" },
+                { obj3, authenticatorConfig3, sequenceConfig3, false, "" }, {
+                        obj4, authenticatorConfig4, sequenceConfig4, true,
+                        X509CertificateConstants.X509_CERTIFICATE_SUBJECTDN_REGEX_NO_MATCHES_ERROR
+                }, {
+                        obj4, authenticatorConfig5, sequenceConfig4, false, ""
+                },
+                {
+                        obj6, authenticatorConfig6, sequenceConfig4, false, ""
+                },
+                };
+    }
```

Comment:
Let's use consistent formatting for all objects.

Reply:
done